sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageBody.java

```diff
@@ -107,10 +141,58 @@ public AmqpMessageBodyType getBodyType() {
     public byte[] getFirstData() {
         if (bodyType != AmqpMessageBodyType.DATA) {
             throw logger.logExceptionAsError(new IllegalArgumentException(
-                "This method can only be called for AMQP Data body type at present. Track this issue, "
-                    + "https://github.com/Azure/azure-sdk-for-java/issues/17614 for other body type support in "
-                    + "future."));
+                String.format(Locale.US, "This method can be called if AMQP Message body type is 'DATA'. "
+                    + "The actual type is [%s].", bodyType)));
         }
         return data;
     }
+
+    /**
+     * Gets the AMQP Value set on this {@link AmqpMessageBody}. It can be any premitive AMQP Data types.
+     *
+     * <p><b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method. Get methods not
+     * corresponding to the type of the body throws exception.</b></p>
+     *
+     * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+     * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+     * @return data set on {@link AmqpMessageBody}.
+     *
+     * @throws IllegalArgumentException If {@link AmqpMessageBodyType} is not {@link AmqpMessageBodyType#DATA DATA}.
+     * @see <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format" target="_blank">
+     *     Amqp Message Format.</a>
+     */
+    public Object getValue() {
+        if (bodyType != AmqpMessageBodyType.VALUE) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                String.format(Locale.US, "This method can be called if AMQP Message body type is 'VALUE'. "
+                        + "The actual type is [%s].", bodyType)));
+        }
+        return value;
+    }
+
+    /**
+     * Gets the unmodifiable AMQP Sequence set on this {@link AmqpMessageBody}.It support only one {@code sequence} at
+     * present.
+     *
+     * <p><b>Client should test for {@link AmqpMessageBodyType} before calling corresponding get method. Get methods not
+     * corresponding to the type of the body throws exception.</b></p>
+     *
+     * <p><strong>How to check for {@link AmqpMessageBodyType}</strong></p>
+     * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+     * @return data set on {@link AmqpMessageBody}.
+     *
+     * @throws IllegalArgumentException If {@link AmqpMessageBodyType} is not {@link AmqpMessageBodyType#DATA DATA}.
+     * @throws UnsupportedOperationException If you try to modify returned sequence.
```

Comment:
This method doesn't throw this exception. So, it must be removed from the javadoc.

Reply:
it was needed  because we return unmodifiable List, and wanted to warn user about it. But I will remove it.